src/components/plots/gridPlot.ts

```diff
@@ -56,14 +61,30 @@ export module Plot {
 
     protected _generateAttrToProjector() {
       var attrToProjector = super._generateAttrToProjector();
-      var xStep = (<Scale.Category> this._xScale).rangeBand();
-      var yStep = (<Scale.Category> this._yScale).rangeBand();
-      attrToProjector["width"]  = () => xStep;
-      attrToProjector["height"] = () => yStep;
-      var xAttr = attrToProjector["x"];
-      var yAttr = attrToProjector["y"];
-      attrToProjector["x"] = (d, i, u, m) => xAttr(d, i, u, m) - xStep / 2;
-      attrToProjector["y"] = (d, i, u, m) => yAttr(d, i, u, m) - yStep / 2;
+
+      // Copy each of the different projectors
+      var x1Attr = attrToProjector["x"];
+      var y1Attr = attrToProjector["y"];
+      var x2Attr = attrToProjector["x2"];
+      var y2Attr = attrToProjector["y2"];
+
+      // Adjust the xScale if it is ordinal
+      if (this._xScale instanceof Scale.Category) {
+        attrToProjector["width"] = () => (<Scale.Category> this._xScale).rangeBand();
+        attrToProjector["x"] = (d, i, u, m) => x1Attr(d, i, u, m) - attrToProjector["width"](d, i, u, m) / 2;
+      } else {
+        attrToProjector["width"] = (d, i, u, m) => Math.abs(x2Attr(d, i, u, m) - x1Attr(d, i, u, m));
```

Comment:
On a curious note, would it be possible to not need to specify x2 for quantitative scale but have more fine-grained control if they do specify it?